lib/matplotlib/legend.py

```diff
@@ -603,11 +592,9 @@ def val_or_rc(val, rc_name):
             'markeredgecolor': ['get_markeredgecolor', 'get_edgecolor'],
             'mec':             ['get_markeredgecolor', 'get_edgecolor'],
         }
+        labelcolor = mpl._val_or_rc(labelcolor, 'legend.labelcolor')
```

Comment:
You could nest another `mpl._val_or_rc(..., 'text.color')` here and add a short comment about falling back to `text.color` if `legend.labelcolor` is not present.